docs/backend_guide.md

```diff
@@ -0,0 +1,227 @@
+        receiver_id (Foreign Key referencing User Table)
+        amount
+        date_created
+
+This schema includes tables for users, transactions, and transaction history. The user table stores user information, including username, email, and password. The transaction table tracks payment transactions between users, including sender, receiver, and amount. The transaction history table records the status and update timestamps for each transaction. Keep in mind that this is a basic example, and the actual database schema for the Power Pay App may require additional tables and relationships. 
+
+
+# Authentication
+
+    In the backend, the authentication process for smartphone and Choronko users in the Power Payment system are handled using specific mechanisms and workflows.
+
+        1. Smartphone User Authentication:
+
+            * User Registration: When a smartphone user registers, they provide their phone number, which is stored in the user database.
+            * OTP Generation and Delivery: Upon registration, an OTP is generated and sent to the user's phone number via SMS using an SMS gateway or service provider.
+            * OTP Verification: The backend validates the OTP entered by the user against the one stored in the database. If the OTP matches, the user is considered authenticated.
+            * PIN Setup: After successful OTP verification, the user is prompted to set a PIN for subsequent authentication. The PIN is securely stored in the user database, preferably using hashing and salting techniques.
+            
+        2. Choronko User Authentication:
+
+            * SMS Gateway Integration: The backend integrates with an SMS gateway to receive and interpret messages sent by Choronko users.
+            * Syntax Interpretation: The SMS gateway extracts the phone number and PIN from the incoming message using the specified syntax, such as numberpin#.
+            * PIN Verification: The backend verifies the received PIN against the one stored in the Choronko user's database record. If the PIN matches, the user is considered authenticated.
+
+
+# Error Handling
+
+    Error handling is an essential aspect of any application, including the Power Payment Backend, to ensure smooth operation and a positive user experience. Proper error handling involves identifying, capturing, and appropriately responding to errors that may occur during the execution of backend processes. Below is an explanation of how error handling should be done in the backend, along with examples:
+
+    - HTTP Status Codes:
+
+        The app utilizes appropriate HTTP status codes to indicate the nature of the error, allowing clients to understand the response at a glance.
+
+        For example, a 400 status code can be used for client-side errors (e.g., invalid request parameters), while a 500 status code can be used for server-side errors (e.g., internal server errors).
+
+    - Error Logging and Notification:
```

Comment:
Good point. You should also indicate how this logging is done, if it's going to include another component in our system. E.g: tools like rollbar or sentry are the external tools used for logging and error handling and SHOULD be documented. Same for the notifications.